crates/red_knot_test/src/parser.rs

```diff
@@ -136,12 +137,13 @@ pub(crate) struct EmbeddedFile<'s> {
 /// Matches an arbitrary amount of whitespace (including newlines), followed by a sequence of `#`
 /// characters, followed by a title heading, followed by a newline.
 static HEADER_RE: LazyLock<Regex> =
-    LazyLock::new(|| Regex::new(r"^(\s*\n)*(?<level>#+)\s+(?<title>.+)\s*\n").unwrap());
+    LazyLock::new(|| Regex::new(r"^(?<level>#+)\s+(?<title>.+)\s*\n").unwrap());
 
 /// Matches a code block fenced by triple backticks, possibly with language and `key=val`
 /// configuration items following the opening backticks (in the "tag string" of the code block).
 static CODE_RE: LazyLock<Regex> = LazyLock::new(|| {
-    Regex::new(r"^```(?<lang>\w+)(?<config>( +\S+)*)\s*\n(?<code>(.|\n)*?)\n?```\s*\n").unwrap()
+    Regex::new(r"^```(?<lang>(?-u:\w)+)(?<config>( +\S+)*)\s*\n(?<code>(.|\n)*?)\n```\s*\n")
```

Comment:
The new regex does not match empty code blocks such as
````markdown
```py path=a.py
```
````
in `mdtest/import/errors.md`, which is also why some tests are failing. The final newline was optional before: `\n?`.

Reply:
It might be worth adding a specific unit test for this edge case in the `red_knot_test` crate, so that we still have tests that will fail even if the mdtest in `mdtest/import/errors.md` changes so it no longer has an empty code block

Reply:
Agree, I'll follow up, once I have @carljm approval